### PR TITLE
Remove usage of json! macro from websocket message sending

### PR DESCRIPTION
**Pull request by GnomedDev**
This also implements https://github.com/discord/discord-api-docs/commit/6755c355caaf2e1c321dc7de17e1e257d9aa2db5

### CODE REVIEW

File: src/gateway/ws.rs [hunk 1]
```diff
@@ -15,12 +15,75 @@ use url::Url;
...
+    limit: u16,
+}
+
+#[derive(Serialize)]
+#[serde(untagged)]
+enum WebSocketMessageData<'a> {
+    Heartbeat(Option<u64>),
+    ChunkGuild(ChunkGuildMessage<'a>),
+    Identify {
+        v: u8,
```
Comment (vaporoxx) [inline src/gateway/ws.rs]: This doesn't actually exist anymore

File: src/gateway/ws.rs [hunk 2]
```diff
@@ -15,12 +15,75 @@ use url::Url;
...
+        session_id: &'a str,
+        token: &'a str,
+        seq: u64,
+    },
+}
+
+#[derive(Serialize)]
+struct WebSocketMessage<'a> {
+    op: OpCode,
+    #[serde(borrow)]
```
Comment (vaporoxx) [inline src/gateway/ws.rs]: From looking at the serde docs it seems like `serde(borrow)` is used for deserialization only, which doesn't apply here (correct me if i'm wrong)

File: src/gateway/ws.rs [hunk 3]
```diff
@@ -112,65 +175,63 @@ impl WsClient {
...
-
-        match filter {
-            ChunkGuildFilter::None => payload["d"]["query"] = json!(""),
-            ChunkGuildFilter::Query(query) => payload["d"]["query"] = json!(query),
-            ChunkGuildFilter::UserIds(user_ids) => {
-                let ids = user_ids.iter().map(|x| x.0).collect::<Vec<u64>>();
-                payload["d"]["user_ids"] = json!(ids);
-            },
+        let (query, user_ids) = match filter {
+            ChunkGuildFilter::None => (None, None),
```
Comment (vaporoxx) [inline src/gateway/ws.rs]: Discord docs state that either `query` or `user_ids` is required, so this should return an empty string for the query instead

File: src/gateway/ws.rs [hunk 4]
```diff
@@ -184,19 +239,19 @@ impl WsClient {
...
-                "afk": false,
-                "since": now,
-                "status": status.name(),
-                "game": activity.as_ref().map(|x| json!({
-                    "name": x.name,
-                    "type": x.kind,
-                    "url": x.url,
-                })),
+        self.send_json(&WebSocketMessage {
+            op: Opcode::RequestGuildMembers,
```
Comment (vaporoxx) [inline src/gateway/ws.rs]:
```suggestion
            op: Opcode::PresenceUpdate,
```
Small mistake from fixing conflicts

**Review by arqunis — APPROVED** [review]

**Pull request merged**